modules/temp-channels/channel-settings.js

```diff
@@ -0,0 +1,253 @@
+                await vchann.permissionOverwrites.create(interaction.guild.members.cache.get(user), {'CONNECT': true});
+            }
+        }
+        interaction.editReply(await embedType(moduleConfig['modeSwitched'], {'%mode%': 'private'}, {ephemeral: true}));
+    }
+
+    vc.isPublic = publicTemp;
+    await vc.save;
+};
+
+/**
+ * @param interaction
+ * @param callerInfo
+ * @returns {Promise<void>}
+ */
+module.exports.userAdd = async function (interaction, callerInfo) {
+    const moduleConfig = interaction.client.configurations['temp-channels']['config'];
+    const vc = await client.models['temp-channels']['TempChannel'].findOne({
+        where: {
+            [Op.and]: [
+                {id: interaction.member.voice.channelId},
+                {creatorID: interaction.member.id}
+            ]
+        }
+    });
+    let allowedUsers = vc.allowedUsers;
+    let addedUser = null;
+    if (callerInfo === 'command') {
+        addedUser = interaction.options.getUser('user');
+    }
+    if (callerInfo === 'modal') {
+        const addedUserString = interaction.fields.getTextInputValue('add-modal-input');
+        try {
+            addedUser = interaction.guild.members.cache.find(member => member.user.tag === addedUserString).user;
+        } catch (e) {
+            try {
+                addedUser = await client.users.fetch(addedUserString);
+            } catch {
+
+            }
+        }
+    }
+
+    if (allowedUsers === '') {
+        allowedUsers = addedUser.id;
+    } else {
+        allowedUsers = allowedUsers + ',' + addedUser.id;
+    }
+    vc.allowedUsers = allowedUsers;
+    await vc.save();
+    const vchann = interaction.guild.channels.cache.get(vc.id);
+    if (!await vchann.permissionsFor(vchann.guild.roles.everyone).has('CONNECT')) {
+        await vchann.permissionOverwrites.create(addedUser, {'CONNECT': true});
+    }
+    await interaction.editReply(await embedType(moduleConfig['userAdded'], {'%user%': addedUser.tag}, {ephemeral: true}));
+};
+
+/**
+ *
+ * @param interaction
+ * @param callerInfo
+ * @returns {Promise<void>}
+ */
+module.exports.userRemove = async function (interaction, callerInfo) {
+    const moduleConfig = interaction.client.configurations['temp-channels']['config'];
+    const vc = await client.models['temp-channels']['TempChannel'].findOne({
+        where: {
+            [Op.and]: [
+                {id: interaction.member.voice.channelId},
+                {creatorID: interaction.member.id}
+            ]
+        }
+    });
+    let allowedUsers = vc.allowedUsers.split(',');
+    let removedUser = null;
+    if (callerInfo === 'command') {
+        removedUser = interaction.options.getUser('user');
+    }
+    if (callerInfo === 'modal') {
+        const removedUserString = interaction.fields.getTextInputValue('remove-modal-input');
+        try {
+            removedUser = interaction.guild.members.cache.find(member => member.user.tag === removedUserString).user;
```

Comment:
If I enter an invalid tag, an error gets thrown and the interaction is not answered, because you try accessing the property "user" of an object that might be undefined

Reply:
Same for addUser

Reply:
This is not resolved yet, a crash occurs with invalid input